### PR TITLE
Initial release of walkthrough / samples

### PR DESCRIPTION
**Pull request by amenarde**

### CODE REVIEW

**Comment by amenarde** [comment]
Not ready for merge, needs

- Cleaned ARM templates
- Sample Applications and packages
- Walkthrough on how the samples work

File: README.md [hunk 1]
```diff
@@ -7,12 +7,12 @@ products:
...
 
-author: athinanthny
-ms.author: atsenthi
+author: athinanthny, amenarde
+ms.author: atsenthi, anmenard
 service: service-fabric
 ---
 
-# Code samples using Managed Identity with Service Fabric
+# Getting started with Managed Identity on Service Fabric
```
Comment (dragav) [inline README.md]: "..for Service Fabric applications"
(We may add support for cluster managed identities.)

File: README.md [hunk 2]
```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
 Taxonomies for products and languages: https://review.docs.microsoft.com/new-hope/information-architecture/metadata/taxonomies?branch=master
 -->
 
-Service Fabric can leverage the Managed Identity for Azure Active Directory (AD) to manage credentials in your application. These code examples show how to do so.
+This walkthrough steps through the process of deploying a service fabric cluster to Azure with managed identity enabled, and then deploying an application that uses managed identity to that cluster.
```
Comment (dragav) [inline README.md]: This sample walks through, or this document walks through, or this is a walk through the process

File: README.md [hunk 3]
```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
 Taxonomies for products and languages: https://review.docs.microsoft.com/new-hope/information-architecture/metadata/taxonomies?branch=master
 -->
 
-Service Fabric can leverage the Managed Identity for Azure Active Directory (AD) to manage credentials in your application. These code examples show how to do so.
+This walkthrough steps through the process of deploying a service fabric cluster to Azure with managed identity enabled, and then deploying an application that uses managed identity to that cluster.
+
+## Prerequisites
+
+- [You will need Powershell and the Az Powershell library to run this sample.](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps)
+- You should have an Azure subscription and resource group you can write to. There is no way to test this sample locally.
```
Comment (dragav) [inline README.md]: This sample requires access to an Azure subscription, as well as required privileges and roles to create and manage resources, as specified below.

File: README.md [hunk 4]
```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
+
+- [You will need Powershell and the Az Powershell library to run this sample.](https://docs.microsoft.com/en-us/powershell/azure/install-az-ps)
+- You should have an Azure subscription and resource group you can write to. There is no way to test this sample locally.
+
+## Overview
+
+### Create a User-Assigned Managed Identity and Key Vault
+
+1. [Create a user assigned managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal)
+    - Make sure you create it in the region you would like your cluster to be in
```
Comment (dragav) [inline README.md]: Maybe it's me, but I prefer a less colloquial tone/form of addressing in official documentation. Prefer describing the limitation/restriction, rather than addressing the user directly in an imperative manner.

File: README.md [hunk 5]
```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
+## Walkthrough
+
+### Step 2: Create a Key Vault and necessary cluster server and client certificates
+<!---
+TODO: Provide an ARM template and deploy script to do this automatically
+--->
+#### Create Key Vault
+1. Under your resource group, click `+ Add`, choose `Key Vault`, and click `Create`
+2. Under `Basics`, fill out parameters, putting your keyvault in the same region that you would like your cluster to be in
+3. Under `Access Policy`, check `Azure Virtual Machines for Deployment`, this is what allows ARM to access to keyvault to get the cluster certificate
```
Comment (dragav) [inline README.md]: ".. this allows the provisioning of the cluster certificate during the VMSS deployment."

File: README.md [hunk 6]
```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
+4. Under `Access Policy`, click `+ Add Access Policy`
+    - Under `Secret Permissions`, check `Get` and `List`
+    - Under `Select Principal`, search and choose the managed identity you creared in Step 2
```
Comment (dragav) [inline README.md]: (typo) created

File: README.md [hunk 7]
```diff
@@ -22,7 +22,82 @@ Guidance on onboarding samples to docs.microsoft.com/samples: https://review.doc
...
+    - It should look like the below:
```
Comment (dragav) [inline README.md]: "Here is a sample access policy"

**Review by dragav — COMMENTED** [review]
Thanks for driving this. My inclination would be to start from the specifics of SF deployments/configuration, and add guidance for other resources (such as creating a vault, managing access, creating storage accounts) only if the public asks for it. There is ample documentation (with samples) on those other Azure resources.

File: README.md (hunk 1 above)
Comment (amenarde) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (amenarde) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (amenarde) [reply README.md]: I agree, I've reworded most of it to be object first (Key Vault should have) rather than user first (you should configur eKV)

File: MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<ApplicationManifest xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" ApplicationTypeName="MISampleAppType" ApplicationTypeVersion="1.0.0" xmlns="http://schemas.microsoft.com/2011/01/fabric">
+  <Parameters>
+    <Parameter Name="MISampleConsole_InstanceCount" DefaultValue="-1" />
```
Comment (dragav) [inline MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml]: This makes the service run on every node in the cluster. That makes it more likely to hit a throttling condition. Did you intend to have this sample run as a -1 service? (If so, why?)

File: MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  <!-- Import the ServiceManifest from the ServicePackage. The ServiceManifestName and ServiceManifestVersion 
+       should match the Name and Version attributes of the ServiceManifest element defined in the 
+       ServiceManifest.xml file. -->
+  <ServiceManifestImport>
+    <ServiceManifestRef ServiceManifestName="MISampleConsolePkg" ServiceManifestVersion="1.0.0" />
+    <ConfigOverrides />
+    <Policies>
+      <ContainerHostPolicies CodePackageRef="Code" Isolation="hyperv" RunInteractive="true">
+        <!-- See https://aka.ms/I7z0p9 for how to encrypt your repository password -->
+        <RepositoryCredentials AccountName="my cr username" Password="my cr password" PasswordEncrypted="false" />
```
Comment (dragav) [inline MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml]: I would suggest adding some invalid characters here, so that it's clear to the user these values are not meant to work out of the box.

File: MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        <!-- < PortBinding ContainerPort="81" EndpointRef="MISampleConsoleTypeEndpoint" /> -->
```
Comment (dragav) [inline MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml]: You can remove commented entries (meaning those which aren't comments/explanatory.)

File: MISampleApp/MISampleApp/ApplicationParameters/Cloud.xml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Application xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="fabric:/MISampleApp" xmlns="http://schemas.microsoft.com/2011/01/fabric">
+  <Parameters>
+    <Parameter Name="MISampleConsole_InstanceCount" Value="-1" />
```
Comment (dragav) [inline MISampleApp/MISampleApp/ApplicationParameters/Cloud.xml]: Same comment for instance count.

File: MISampleApp/MISampleApp/PublishProfiles/Cloud.xml [hunk 12]
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<PublishProfile xmlns="http://schemas.microsoft.com/2015/05/fabrictools">
```
Comment (dragav) [inline MISampleApp/MISampleApp/PublishProfiles/Cloud.xml]: Are these files (under PublishProfiles) required?

File: MISampleApp/MISampleWeb/ServiceEventSource.cs [hunk 13]
```diff
@@ -0,0 +1,190 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
```
Comment (dragav) [inline MISampleApp/MISampleWeb/ServiceEventSource.cs:3]: Is this file really necessary? IMO the point of a sample is a minimal but complete-for-purpose exemplification of using a given feature. Everything else just adds noise.

File: MISampleApp/SFMISampleConsole/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,3 @@
+FROM mcr.microsoft.com/windows/nanoserver:1809
```
Comment (dragav) [inline MISampleApp/SFMISampleConsole/Dockerfile:3]: We should add a comment here showing what are compatible Win OS images that can support this type of container. Ideally a link to the compatibility doc (https://docs.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility)

File: MISampleApp/SFMISampleConsole/run.ps1 [hunk 15]
```diff
@@ -0,0 +1,11 @@
+param([string]$mode="env:cert")
+
+# Docker image name for the application
```
Comment (dragav) [inline MISampleApp/SFMISampleConsole/run.ps1:3]: This file can be left out, it's for local testing purposes. (And won't run without the SF-supplied env vars.)

File: MISampleApp/VaultProbe/VaultProbe.cs [hunk 16]
```diff
@@ -0,0 +1,221 @@
...
+
+            var requestUri = $"{config.ManagedIdentityEndpoint}?api-version={config.ManagedIdentityApiVersion}&resource={encodedResource}";
+            Log(LogLevel.Verbose, $"request uri: {requestUri}");
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Get, requestUri);
+            requestMessage.Headers.Add(config.ManagedIdentityAuthenticationHeader, config.ManagedIdentityAuthenticationCode);
+            Log(LogLevel.Verbose, $"added header '{config.ManagedIdentityAuthenticationHeader}': '{config.ManagedIdentityAuthenticationCode}'");
+
+            var response = await httpClient.SendAsync(requestMessage)
+                .ConfigureAwait(false);
```
Comment (dragav) [inline MISampleApp/VaultProbe/VaultProbe.cs:168]: We should add here support for throttling responses. Maybe not in the first iteration, but let's file a bug/GH issue to address it.

File: src/SFMISampleWeb/Program.cs [hunk 17]
```diff
@@ -0,0 +1,26 @@
+﻿namespace Azure.ServiceFabric.ManagedIdentity.Samples
```
Comment (dragav) [inline src/SFMISampleWeb/Program.cs]: Is this duplicated in the sample?

File: src/VaultProbe/ManagedIdentityTokenResponse.cs [hunk 18]
```diff
@@ -0,0 +1,23 @@
+﻿namespace Azure.ServiceFabric.ManagedIdentity.Samples
```
Comment (dragav) [inline src/VaultProbe/ManagedIdentityTokenResponse.cs]: Is this duplicated?

File: MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml (hunk 9 above)
Comment (amenarde) [reply MISampleApp/MISampleApp/ApplicationPackageRoot/ApplicationManifest.xml]: resolved

File: MISampleApp/VaultProbe/VaultProbe.cs (hunk 16 above)
Comment (amenarde) [reply MISampleApp/VaultProbe/VaultProbe.cs:168]: will create issue
